plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -1760,6 +1760,40 @@ private TableStatistics getTableStatistics(String tableName, Constraint constrai
                 });
     }
 
+    @Test
+    public void testPartitionValuesWithSpecialCharacters()
+    {
+        assertUpdate("CREATE TABLE iceberg.tpch.test_special_characters (col_1 INTEGER, col_2 VARCHAR) WITH (partitioning = ARRAY['col_2'])");
+        String values = "" +
+                "(1, 'abcd ef'), " +
+                "(2, 'zip per'), " +
+                "(3, 'race car'), " +
+                "(4, 'with:colon'), " +
+                "(5, 'with-hyphen'), " +
+                "(6, 'with?question'), " +
+                "(7, ' startof'), " +
+                "(8, 'endof '), " +
+                "(9, 'with👨‍🏭factoryman'), " +
+                "(10, ' 👨‍🏭'), " +
+                "(11, '👨‍🏭 '), " +
+                "(12, 'with!exclamation'), " +
+                "(13, 'with.dot'), " +
+                "(14, 'with/slash'), " +
+                "(15, 'with%percent'), " +
+                "(16, 'with$dollar'), " +
+                "(17, 'with#pound'), " +
+                "(18, 'with@at'), " +
+                "(19, 'with*star'), " +
+                "(20, 'with&amp'), " +
+                "(21, 'with=equals'), " +
+                "(22, 'with\\backslash')," +
+                "(23, 'with€euro')";
```

Comment:
€ is not as interesting as $ (since engineers assign certain ascii symbols a special meaning, but € is not ascii),
but € is a good example of a non-ascii

maybe add `with non-ascii ąęłóść ä...` too?